### ISSUE TITLE
DeserializationError while listing Data Factory Linked Services

### ISSUE THREAD

**Issue by flvndh** [Azure/azure-sdk-for-python#7688]
Hello,

While trying to run this piece of code to list a factory linked services:

```python
import os

from azure.common.client_factory import get_client_from_cli_profile
from azure.mgmt.datafactory import DataFactoryManagementClient


RESOURCE_GROUP_NAME = os.environ["RESOURCE_GROUP_NAME"]
FACTORY_NAME = os.environ["FACTORY_NAME"]

client = get_client_from_cli_profile(DataFactoryManagementClient)
linked_services = list(client.linked_services.list_by_factory(RESOURCE_GROUP_NAME, FACTORY_NAME))
```

I get a `msrest.exceptions.DeserializationError` with this stacktrace:
```
Traceback (most recent call last):
  File "...\site-packages\msrest\serialization.py", line 1293, in _deserialize
    found_value = key_extractor(attr, attr_desc, data)
  File "...\site-packages\msrest\serialization.py", line 1064, in rest_key_extractor
    return working_data.get(key)
AttributeError: 'str' object has no attribute 'get'
```

The error seems to occur while trying to deserialize this template (values obfuscated):
```json
{
    "id": "/subscriptions/{subscription_id}/resourceGroups/{resource_group_name}/providers/Microsoft.DataFactory/factories/{factory_name}/linkedservices/{linked_service_name}",
    "name": "{linked_service_name}",
    "type": "Microsoft.DataFactory/factories/linkedservices",
    "properties": {
        "annotations": [],
        "type": "PostgreSql",
        "typeProperties": {
            "connectionString": "Host={hostname};Port={port};Database={database};UID={uid};EncryptionMethod={encryption_method}",
            "password": {
                "type": "AzureKeyVaultSecret",
                "store": {
                    "referenceName": "{key_vault_linked_service_name}",
                    "type": "LinkedServiceReference"
                },
                "secretName": "{secret_name}"
            }
        },
        "connectVia": {
            "referenceName": "{integration_runtime_name}",
            "type": "IntegrationRuntimeReference"
        }
    },
    "etag": "{etag}"
}
```

Do you have any clue on what's going on here ?

**Comment by kaerm** [comment]
@flvndh thanks for letting us know! I am tagging the right team to take a look at this

**Comment by BigCat20196** [comment]
Datafactory has released the latest version `2.2.0`. this problem has been solved. Thank you.

Fyi:
https://pypi.org/project/azure-mgmt-datafactory/2.2.0/
